Sources/BrowserServicesKit/ContentBlocking/ContentBlockingRulesCompilationTask.swift

```diff
@@ -81,9 +112,14 @@ extension ContentBlockerRulesManager {
 
         private func compilationSucceeded(with compiledRulesList: WKContentRuleList,
                                           model: ContentBlockerRulesSourceModel,
+                                          resultType: CompilationResult.ResultType,
                                           completionHandler: @escaping Completion) {
+
+            self.result = self.getCompilationResult(ruleList: compiledRulesList,
+                                                    model: model,
+                                                    resultType: resultType)
+
             workQueue.async {
-                self.result = (compiledRulesList, model)
```

Comment:
I think we should be assigning this back here, as access to `result` should be encapsulated in the queue.